deltas/tokenizers/wikitext_split.py

```diff
@@ -97,4 +186,4 @@
     ("etc", r"."),
 ]
 
-wikitext_split = RegexTokenizer(LEXICON)
+wikitext_split = RegexTokenizer(LEXICON, CJK_LEXICON)
```

Comment:
It looks like tokenizing using CJK is substantially slower.  I think we'll want to have a separate tokenizers for CJK.  It's nice to have a fully parameterized import.  I'm thinking we want something like:

```
from deltas.tokenizers import wikitext_split
from deltas.tokenizers import wikitext_split_w_cjk
```

I think I'd like to have a flexible way to design a tokenization pipeline.  E.g.:

```python
class TokenizerPipeline(Tokenizer):

  def __init__(self, tokenizer, token_processors*):
    self.tokenizer = tokenizer  # start of the pipeline
    self.token_processors = token_processor

  def tokenize(self, text):
    tokens = self.tokenizer.tokenize(text)
    for token_processor in self.token_processors:
      tokens = token_processor.process(tokens)

class TokenProcessor:
  def process(self, tokens):
    raise NotImplementedError()

class CJKSegmenter:
  def process(self, tokens):
    # do the logic of looking for CJK sequences and segmenting them here. 

wikitext_split_w_cjk = TokenizerPipeline(wikitext_split, CJKSegmenter())
```

What do you think of that pattern?

Reply:
I adjusted the code per your comment, now I think it starts to reflect your initial idea to "create tokenization strategy". Please check it and let me know what you think. Example usage for tokenization of JP haiku page:
```
import mwapi
import deltas
import deltas.tokenizers

session = mwapi.Session("https://ja.wikipedia.org")
doc = session.get(action="query", prop="revisions",
                  titles="俳句", rvprop="content", rvslots="main",
                  formatversion=2)
text = doc['query']['pages'][0]['revisions'][0]['slots']['main']['content']

tokenized_text = deltas.tokenizers.wikitext_split.tokenize(text)
tokenized_text_cjk = deltas.tokenizers.wikitext_split_w_cjk.tokenize(text)
```
if you think it's ok we can create an example page and ask someone to look at the tokenized result, if it makes sense in Chinese/Japanse/Korean...